python/plugins/processing/gui/Postprocessing.py

```diff
@@ -324,7 +324,10 @@ def handleAlgorithmResults(
         if project and insertion_point:
             project.layerTreeRegistryBridge().setLayerInsertionPoint(insertion_point)
 
-        project.addMapLayer(layer_node.layer())
+        added_layer = project.addMapLayer(layer_node.layer())
```

Comment:
I'm not sure why all the `if project` are needed, anyway, if they are needed, shouldn't also `added_layer = project.addMapLayer(layer_node.layer())` be inside an `if project`?

Reply:
I don't know. This logic was introduced by 0180380076f8fcdff8c1f58f5d5789a57d8c5cb9 to fix an other regression. Maybe @nyalldawson will have some insights.